json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -181,3 +181,28 @@ protected function result() {
 		return Jetpack_Sync_Settings::get_settings();
 	}
 }
+
+// GET /sites/%s/sync-object
+class Jetpack_JSON_API_Sync_Object extends Jetpack_JSON_API_Sync_Endpoint {
+	protected function result() {
+		$args = $this->query_args();
+
+		$module_name = $args['module_name'];
+
+		require_once dirname( __FILE__ ) . '/../../sync/class.jetpack-sync-modules.php';
+
+		if ( ! $sync_module = Jetpack_Sync_Modules::get_module( $module_name ) ) {
+			return new WP_Error( 'invalid_module', 'You specified an invalid sync module' );
+		}
+
+		$object_type = $args['object_type'];
+		$object_id   = $args['object_id'];
+
+		require_once dirname( __FILE__ ) . '/../../sync/class.jetpack-sync-sender.php';
+		$codec = Jetpack_Sync_Sender::get_instance()->get_codec();
```

Comment:
I thought about that, but the object has to be encoded _somehow_, so we have to either select:

- json (throws away important information we need, like array vs object
- PHP serialize (dangerous) 
- yaml? we don't use it anywhere, may have the same issue as json
- ...

So I figured, if it needs to be encoded, probably best to use the default encoder - unless you have a reason not to?